diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2e3a686be..642f0c2c9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -213,8 +218,8 @@ target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC
 target_link_libraries(${PROJECT_NAME} PUBLIC ${Boost_LIBRARIES})
 
 # We expect find_package(Ceres) to have located these for us.
-target_link_libraries(${PROJECT_NAME} PUBLIC glog)
-target_link_libraries(${PROJECT_NAME} PUBLIC gflags)
+#target_link_libraries(${PROJECT_NAME} PUBLIC glog)
+#target_link_libraries(${PROJECT_NAME} PUBLIC gflags)
 
 target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC
   "${CAIRO_INCLUDE_DIRS}")
diff --git a/cartographer/common/configuration_file_resolver.cc b/cartographer/common/configuration_file_resolver.cc
index 9774caf10..99c703c7f 100644
--- a/cartographer/common/configuration_file_resolver.cc
+++ b/cartographer/common/configuration_file_resolver.cc
@@ -19,6 +19,7 @@
 #include <fstream>
 #include <iostream>
 #include <streambuf>
+#include <cstdlib>
 
 #include "cartographer/common/config.h"
 #include "glog/logging.h"
@@ -30,6 +31,8 @@ ConfigurationFileResolver::ConfigurationFileResolver(
     const std::vector<std::string>& configuration_files_directories)
     : configuration_files_directories_(configuration_files_directories) {
   configuration_files_directories_.push_back(kConfigurationFilesDirectory);
+  if(const char* config_dir = std::getenv("CARTOGRAPHER_CONFIG_DIR"))
+    configuration_files_directories_.push_back(config_dir);
 }
 
 std::string ConfigurationFileResolver::GetFullPathOrDie(

diff --git a/cartographer/mapping/pose_graph_interface.h b/cartographer/mapping/pose_graph_interface.h
index 6483ef4e9..645672077 100644
--- a/cartographer/mapping/pose_graph_interface.h
+++ b/cartographer/mapping/pose_graph_interface.h
@@ -18,6 +18,7 @@
 #define CARTOGRAPHER_MAPPING_POSE_GRAPH_INTERFACE_H_
 
 #include <vector>
+#include <array>
 
 #include "cartographer/common/optional.h"
 #include "cartographer/mapping/id.h"
