diff --git a/CMakeLists.txt b/CMakeLists.txt
index ef2fcb3e69..08dff22544 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,14 +39,8 @@ find_package(Boost REQUIRED COMPONENTS ${BOOST_COMPONENTS})
 find_package(Ceres REQUIRED COMPONENTS SuiteSparse)
 find_package(Eigen3 REQUIRED)
 find_package(LuaGoogle REQUIRED)
-if(WIN32)
-  # On Windows, Protobuf is incorrectly found by the bundled CMake module, so prefer native CMake config.
-  set(protobuf_MODULE_COMPATIBLE TRUE CACHE INTERNAL "")
-  find_package(Protobuf 3.0.0 CONFIG)
-else()
-  find_package(Protobuf 3.0.0 REQUIRED)
-endif()
-
+set(protobuf_MODULE_COMPATIBLE TRUE CACHE INTERNAL "")
+find_package(Protobuf CONFIG REQUIRED)
 if (${BUILD_GRPC})
   find_package(async_grpc REQUIRED)
 endif()
@@ -268,7 +262,7 @@ target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC
   ${PROTOBUF_INCLUDE_DIR})
 # TODO(hrapp): This should not explicitly list pthread and use
 # PROTOBUF_LIBRARIES, but that failed on first try.
-target_link_libraries(${PROJECT_NAME} PUBLIC ${PROTOBUF_LIBRARY} 
+target_link_libraries(${PROJECT_NAME} PUBLIC protobuf::libprotobuf
   absl::algorithm
   absl::base
   absl::debugging
