diff --git a/cartographer/mapping/internal/motion_filter.cc b/cartographer/mapping/internal/motion_filter.cc
index 715b0408f1..525232a9e3 100644
--- a/cartographer/mapping/internal/motion_filter.cc
+++ b/cartographer/mapping/internal/motion_filter.cc
@@ -39,9 +39,9 @@ MotionFilter::MotionFilter(const proto::MotionFilterOptions& options)
 
 bool MotionFilter::IsSimilar(const common::Time time,
                              const transform::Rigid3d& pose) {
-  LOG_IF_EVERY_N(INFO, num_total_ >= 500, 500)
-      << "Motion filter reduced the number of nodes to "
-      << 100. * num_different_ / num_total_ << "%.";
+  //LOG_IF_EVERY_N(INFO, num_total_ >= 500, 500)
+  //    << "Motion filter reduced the number of nodes to "
+  //    << 100. * num_different_ / num_total_ << "%.";
   ++num_total_;
   if (num_total_ > 1 &&
       time - last_time_ <= common::FromSeconds(options_.max_time_seconds()) &&
diff --git a/cartographer/sensor/internal/ordered_multi_queue.cc b/cartographer/sensor/internal/ordered_multi_queue.cc
index 26c8ee16d1..213b21365f 100644
--- a/cartographer/sensor/internal/ordered_multi_queue.cc
+++ b/cartographer/sensor/internal/ordered_multi_queue.cc
@@ -64,8 +64,8 @@ void OrderedMultiQueue::Add(const QueueKey& queue_key,
                             std::unique_ptr<Data> data) {
   auto it = queues_.find(queue_key);
   if (it == queues_.end()) {
-    LOG_EVERY_N(WARNING, 1000)
-        << "Ignored data for queue: '" << queue_key << "'";
+    //LOG_EVERY_N(WARNING, 1000)
+    //    << "Ignored data for queue: '" << queue_key << "'";
     return;
   }
   it->second.queue.Push(std::move(data));
@@ -152,7 +152,7 @@ void OrderedMultiQueue::CannotMakeProgress(const QueueKey& queue_key) {
   blocker_ = queue_key;
   for (auto& entry : queues_) {
     if (entry.second.queue.Size() > kMaxQueueSize) {
-      LOG_EVERY_N(WARNING, 60) << "Queue waiting for data: " << queue_key;
+      //LOG_EVERY_N(WARNING, 60) << "Queue waiting for data: " << queue_key;
       return;
     }
   }
